src/components/select/SelectAsync.tsx

```diff
@@ -109,41 +100,56 @@ export const SelectAsync = forwardRef<HTMLInputElement, PropsWithLabels<Props>>(
     [onChange],
   );
 
+  const itemComponent = useSelectItemComponent(optionComponent, optionType);
+  const isClearable = !isNotClearable && !isRequired;
+
+  // TODO Highlighter for search
+
   return (
-    <MantineSelect
-      allowDeselect={!isRequired}
+    <SelectStyled
+      allowDeselect={isClearable}
       aria-label={ariaLabel}
-      clearable={!isNotClearable && !isRequired}
+      aria-labelledby={ariaLabelledBy}
+      clearable={isClearable}
       data={data}
+      data-variant={highlight}
       description={helpText}
       disabled={isDisabled}
       error={labelError ?? hasError}
       filter={() => true}
-      itemComponent={optionComponent}
+      icon={valueIcon}
+      inputSize={size}
+      itemComponent={itemComponent}
       label={label}
       nothingFound={labelNotFound}
       onChange={handleChange}
+      onDropdownOpen={onOpen}
       onSearchChange={handleSearch}
       ref={ref}
       required={isRequired}
-      rightSection={isLoading ? <Spinner isLoading /> : undefined}
+      rightSection={getSelectRightSection({
+        hasValue: isDefined(value),
+        isLoading,
+        isClearable,
+      })}
       searchable
-      value={value}
+      value={value ?? null} // Mantine only clears the value if `null`, not `undefined`
       variant={highlight}
+      {...selectProps}
     />
   );
```

Comment:
This is mostly shared between the two, I wonder if it isn't better to have a `SelectBase` ?